app/models/run.rb

```diff
@@ -163,6 +163,30 @@ def publish_aging
     publish_age_every(:day, 30)
   end
 
+  def find_vod!
+    return if user.nil? || user.twitch.nil?
+
+    pb = histories.find_by(realtime_duration_ms: realtime_duration_ms, gametime_duration_ms: gametime_duration_ms)
+
+    return if pb.nil?
+
+    Twitch::Videos.recent(user.twitch.twitch_id).each do |vod|
+      match = /^((\d+)h)?((\d+)m)?((\d+)s)?$/.match(vod['duration'])
+      hours   = match[2].to_i.hours
+      minutes = match[4].to_i.minutes
+      seconds = match[6].to_i.seconds
+
+      vod_start = DateTime.parse(vod['created_at'])
+      vod_end   = vod_start + hours + minutes + seconds
+
+      if vod_start - 2.hours < pb.started_at && pb.ended_at < vod_end + 2.hours
```

Comment:
In practice I found videos up to 90 minutes and 30 seconds "off" from the times recorded in the runs, and I don't know why.